client/signup/jetpack-connect/authorize.jsx

```diff
@@ -2,64 +2,113 @@
  * External dependencies
  */
 import React from 'react';
-import Debug from 'debug';
+import store from 'store';
 
 /**
  * Internal dependencies
  */
 import ConnectHeader from './connect-header';
 import Main from 'components/main';
-import wpcom from 'lib/wp';
+import { jetpackAuthorize, createAccount } from './authorize-action';
+import LoggedOutFormLinks from 'components/logged-out-form/links';
+import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
+import SignupForm from 'components/signup-form';
+import WpcomLoginForm from 'signup/wpcom-login-form';
+import _user from 'lib/user';
+import config from 'config';
 
 /**
  * Module variables
  */
-const debug = new Debug( 'calypso:jetpack-connect-authorize' );
+const userModule = _user();
 
-export default React.createClass( {
-	displayName: 'JetpackConnectAuthorize',
+var LoggedOutForm = React.createClass( {
+	displayName: 'LoggedOutForm',
 
 	getInitialState() {
-		return {
-			isSubmitting: false,
-			authorizeError: false,
-			authorizeSuccess: false
+		return { error: false, userData: false, bearerToken: false, isSubmitting: false };
+	},
+
+	submitForm( form, userData ) {
+		this.setState( { submitting: true } );
+
+		const createAccountCallback = ( error, data ) => {
+			if ( error ) {
+				this.setState( { submitting: false } );
+			} else if ( data.bearer_token ) {
+				this.setState( { userData, bearerToken: data.bearer_token } );
+			}
 		};
+
+		store.set( 'jetpack_connect_authorize_after_signup', '1' );
```

Comment:
redux-state persistence should be coming to production really soon, I think that's a better way to handle this. We have reached quotes of `localStorage` quite frequently in the past.